wpsc-components/merchant-core-v3/gateways/worldpay.php

```diff
@@ -0,0 +1,743 @@
+		
+		$order_info = $this->refresh_transaction_info( $wp_transaction_id );
+	
+		$wp_auth_code		= $this->log->get( 'wp_authcode' );
+		$wp_order_status	= $this->log->get( 'wp_order_status' );
+		
+		?>
+		
+		<div class="metabox-holder">
+			<div id="wpsc-worldpay-payments" class="postbox">
+				<h3 class='hndle'><?php _e( 'WorldPay Payments' , 'wp-e-commerce' ); ?></h3>
+				<div class='inside'>
+					<p><?php
+							_e( 'Current status: ', 'wp-e-commerce' );
+							echo wp_kses_data( $this->log->get( 'worldpay-status' ) );
+						?>
+					</p>
+					<p><?php
+							_e( 'Transaction ID: ', 'wp-e-commerce' );
+							echo wp_kses_data( $wp_transaction_id );
+						?>
+					</p>
+		<?php
+		
+		//Show actions based on order status
+		switch ( $wp_order_status ) {
+			case 'Open' :
+				//Order is only authorized and still not captured/voided
+				$actions['capture'] = array(
+					'id' => $wp_transaction_id,
+					'button' => __( 'Capture funds', 'wp-e-commerce' )
+				);
+				
+				//
+				if ( ! $order_info['settled'] ) {
+					//Void
+					$actions['void'] = array(
+						'id' => $wp_transaction_id,
+						'button' => __( 'Void order', 'wp-e-commerce' )
+					);					
+				}
+				
+				break;
+			case 'Completed' :
+				//Order has been captured or its a direct payment
+				if ( $order_info['settled'] ) {
+					//Refund
+					$actions['refund'] = array(
+						'id' => $wp_transaction_id,
+						'button' => __( 'Refund order', 'wp-e-commerce' )
+					);
+				} else {
+					//Void
+					$actions['void'] = array(
+						'id' => $wp_transaction_id,
+						'button' => __( 'Void order', 'wp-e-commerce' )
+					);					
+				}
+				
+			break;
+			case 'Refunded' :
+			case 'Voided' :
+			
+			break;
+		}			
+		
+		if ( ! empty( $actions ) ) {
+
+			echo '<p class="buttons">';
+
+			foreach ( $actions as $action_name => $action ) {
+				echo '<a href="#" class="button" data-action="' . $action_name . '" data-id="' . $action['id'] . '">' . $action['button'] . '</a> ';
+			}
+
+			echo '</p>';
+
+		}		
+		?>
+		<script type="text/javascript">
+		jQuery( document ).ready( function( $ ) {
+			$('#wpsc-worldpay-payments').on( 'click', 'a.button, a.refresh', function( e ) {
+				var $this = $( this );
+				e.preventDefault();
+
+				var data = {
+					action: 		'worldpay_order_action',
+					security: 		'<?php echo wp_create_nonce( "wp_order_action" ); ?>',
+					order_id: 		'<?php echo $order_id; ?>',
+					worldpay_action: 	$this.data('action'),
+					worldpay_id: 		$this.data('id'),
+					worldpay_refund_amount: jQuery('.worldpay_refund_amount').val(),
+				};
+
+				// Ajax action
+				$.post( ajaxurl, data, function( result ) {
+						location.reload();
+					}, 'json' );
+
+				return false;
+			});
+		} );
+
+		</script>
+		</div>
+		</div>
+		</div>
+		<?php
+	}
+
+    /**
+     * Get the order status from API
+     *
+     * @param  string $transaction_id
+     */	
+	public function refresh_transaction_info( $transaction_id ) {
```

Comment:
Scrutinizer is telling us that this (and the method above) are a bit too complex.  Let's see how we may be able to refactor.